projects/igniteui-angular/src/lib/directives/input/input.directive.ts

```diff
@@ -52,6 +52,9 @@ export class IgxInputDirective implements AfterViewInit, OnDestroy {
     @Input('value')
     set value(value: any) {
         this.nativeElement.value = value;
+        if (!this.ngControl && this._hasValidators) {
```

Comment:
This piece of code is repeated again in the [input event handler](https://github.com/IgniteUI/igniteui-angular/pull/3564/files#diff-6f41a12bd84d5e6d3bad3cad109e10c3R150) Can we abstract it away, say as `checkValidity()` that returns an `IgxInputState` rather than copy it as it is?